core/templates/pages/classroom-admin-page/classroom-admin-page.component.ts

```diff
@@ -761,7 +770,21 @@ export class ClassroomAdminPageComponent implements OnInit {
     );
   }
 
-  saveClassroomValidationErrors(): string[] {
+  canSaveClassroom(): boolean {
+    if (
+      (this.tempClassroomData.getIsPublished() &&
+        this.validationErrors.length !== 0) ||
+      this.saveClassroomValidationErrors.length !== 0
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  getSaveClassroomValidationErrors(): string[] {
+    if (!this.canSaveClassroom()) {
+      return this.validationErrors;
```

Comment:
This seems incorrect. Suppose the classroom is not published, you should return here only saveClassroomValidationErrors, no?

This actually indicates a logic duplication. You have one function (canSaveClassroom) that should return true iff another function (getSaveClassroomValidationErrors) returns an  empty list. Logic duplication is bad, for example because you need to synchronize logic between two places. Here, the logic is not in sync.

What if we do like this - getSaveClassroomValidationErrors doesn't use canSaveClassroom, but returns errors depending on the isPublished. Then canHaveClassroom can be implemented simply as getSaveClassroomValidationErrors().length === 0. Would that work?

Reply:
that would be much better. I'll update it.

Reply:
Done.